src/Chapter2.hs

```diff
@@ -348,8 +353,10 @@ Implement a function that returns only the first half of a given list.
 >>> firstHalf "bca"
 "b"
 -}
--- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
+
+firstHalf :: [a] -> [a]
+firstHalf l =
+  let half = fst (divMod (length l ) 2) in take half l
```

Comment:
Great usage of `let-in` construction! Makes code easily readable 💯 

Note, that as you use only the `fst` of `divMod`, you can use `div` function straightaway 🙂